efel/pyfeatures/extrafeats.py

```diff
@@ -0,0 +1,521 @@
+"""Extra features functions"""
+
+"""
+Copyright (c) 2016-2020, EPFL/Blue Brain Project
+
+ This file is part of BluePyOpt <https://github.com/BlueBrain/BluePyOpt>
+
+ This library is free software; you can redistribute it and/or modify it under
+ the terms of the GNU Lesser General Public License version 3.0 as published
+ by the Free Software Foundation.
+
+ This library is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with this library; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
+
+import numpy as np
+
+all_1D_features = [
+    "peak_to_valley",
+    "halfwidth",
+    "peak_trough_ratio",
+    "repolarization_slope",
+    "recovery_slope",
+    "neg_peak_relative",
+    "pos_peak_relative",
+    "neg_peak_diff",
+    "pos_peak_diff",
+    "neg_image",
+    "pos_image",
+]
+
+
+def calculate_features(
+    waveforms,
+    sampling_frequency,
+    upsample=None,
+    feature_names=None,
+    recovery_slope_window=0.7
+):
+    """Calculate features for all waveforms
+
+    Parameters
```

Comment:
The dostrings in this file are more detailed than in other python files. :)  I think we can use this format

Reply:
Does the documentation output looks consistent though? The standard we were using can be found e.g. here: https://github.com/BlueBrain/eFEL/blob/master/efel/api.py#L45

Reply:
it looks like this: 
![image](https://github.com/BlueBrain/eFEL/assets/8694327/bb19fe67-1eff-4771-8fe7-5c2b4058dd74)

Reply:
It would be better if we could have the same standard: In the other files, we are using Args, not Parameters. Also the return block is strange here, part of the sentence is in bold for some reason.

Reply:
is this fine now?: ![image](https://github.com/BlueBrain/eFEL/assets/8694327/cd944299-161b-4ed7-8608-4d89aae9aef2)

Reply:
yes, looks good!